clients/spark/core/src/test/resources/parser-test/sst_files_generator.go

```diff
@@ -0,0 +1,291 @@
+
+	// BlockSize is the target uncompressed size in bytes of each table block.
+	// The default value is 4096.
+	blockSize := 1 + rand.Intn(9000)
+	writerOptions = sstable.WriterOptions{
+		Compression:             sstable.SnappyCompression,
+		TablePropertyCollectors: []func() sstable.TablePropertyCollector{NewStaticCollector(DefaultUserProperties)},
+		BlockSize:               blockSize,
+	}
+	fmt.Printf("BlockSize %d...\n", blockSize)
+	createTestInputFiles(keys, generateNanoid, sizeBytes, "fuzz.block.size", writerOptions)
+
+	// BlockSizeThreshold finishes a block if the block size is larger than the
+	// specified percentage of the target block size and adding the next entry
+	// would cause the block to be larger than the target block size.
+	//
+	// The default value is 90
+	blockSizeThreshold := 1 + rand.Intn(100)
+	writerOptions = sstable.WriterOptions{
+		Compression:             sstable.SnappyCompression,
+		TablePropertyCollectors: getDefaultTablePropertyCollectors(DefaultUserProperties),
+		BlockSizeThreshold:      blockSizeThreshold,
+	}
+	fmt.Printf("BlockSizeThreshold %d...\n", blockSizeThreshold)
+	createTestInputFiles(keys, generateNanoid, sizeBytes, "fuzz.block.size.threshold", writerOptions)
+}
+
+func writeLargeSsts() {
+	sizeBytes := DefaultCommittedPermanentMaxRangeSizeBytes
+	keys := prepareSortedSliceWithNanoid(sizeBytes)
+	writerOptions := sstable.WriterOptions{
+		Compression:             sstable.SnappyCompression,
+		TablePropertyCollectors: getDefaultTablePropertyCollectors(DefaultUserProperties),
+	}
+	createTestInputFiles(keys, generateNanoid, sizeBytes, "max.size.lakefs.file", writerOptions)
+}
+
+func writeMultiSizedSstsWithContentsFuzzing() {
+	const numOfFilesToGenerate = 6
+	for i := 0; i < numOfFilesToGenerate; i++ {
+		var curSize int
+		if i%2 == 0 {
+			curSize = 1 + rand.Intn(DefaultCommittedPermanentMaxRangeSizeBytes/MiBToBytes)*MiBToBytes
+		} else {
+			curSize = 1 + rand.Intn(DefaultMaxSizeKb)*KiBToBytes
+		}
+
+		testFileName := fmt.Sprintf("fuzz.contents.%d", i)
+		keys := prepareSortedSliceWithFuzzing(curSize)
+
+		writerOptions := sstable.WriterOptions{
+			Compression:             sstable.SnappyCompression,
+			TablePropertyCollectors: getDefaultTablePropertyCollectors(DefaultUserProperties),
+		}
+		createTestInputFiles(keys, newGenerateFuzz(), curSize, testFileName, writerOptions)
+	}
+}
+
+func prepareSortedSliceWithFuzzing(size int) []string {
+	var keySumBytes int
+	var slice []string
+	generateFuzz := newGenerateFuzz()
+	// Keep generating keys until we reach the desired slice size. This guarantees that the sstable writer will have
+	// enough data to write an sstable of the desired size.
+	for keySumBytes < size {
+		key, _ := generateFuzz()
+		if key == "" { // TODO(Tals): remove after resolving https://github.com/treeverse/lakeFS/issues/2419
+			continue
+		}
+		keySumBytes += len(key)
+		slice = append(slice, key)
```

Comment:
This will barely go through the prefix decompression code, because keys will have only short shared prefixes.  Consider randomizing a prefix length at each key, then adding fuzz to _approximately_ a constant length.